R/disease_progression.R

```diff
@@ -19,15 +19,27 @@ update_infection <- function(
 }
 
 create_progression_process <- function(
-  state,
-  from_state,
-  to_state,
-  rate,
-  infectivity,
-  new_infectivity
-  ) {
+    state,
+    from_state,
+    to_state,
+    rate,
+    infectivity,
+    new_infectivity
+) {
   function(timestep) {
-    to_move <- state$get_index_of(from_state)$sample(1/rate)
+    
+    # Retrieve the indices of all individuals in the to_move state:
+    index <- state$get_index_of(from_state)
+    
+    # If the length of rate is greater than 1 (when it's a variable):
+    if (length(rate) > 1) {
```

Comment:
```suggestion
    if (inherits(rate , "DoubleVariable")) {
```

This should be equivalent, but it demonstrates the intent more clearly IMO.

Reply:
I've never used this approach before but I agree it's clearer about what we're checking for - will make the change! Thanks @plietar!